src/Confluent.Kafka/Consumer.cs

```diff
@@ -806,16 +807,17 @@ internal Consumer(ConsumerBuilder<TKey, TValue> builder)
             }
             else
             {
-                throw new ArgumentException("Invalid key deserializer configuration.");
+                // enforced by the builder class.
+                throw new Exception("FATAL: Both async and sync key deserializers were set.");
```

Comment:
I was initially using `Trace.Fail` for programmer errors (i.e. things that would only occur if we made a mistake), but that was causing problems on .netframework45.

Reply:
Why use the base Exception here and not InvalidOperation?

Reply:
that's better yes, changing.